pkg/object/ingresscontroller/translator.go

```diff
@@ -317,6 +317,10 @@ func (st *specTranslator) translateTLSConfig(b *httpServerSpecBuilder, ingresses
 }
 
 func (st *specTranslator) translateIngressRules(b *httpServerSpecBuilder, ingress *apinetv1.Ingress) {
+	hostRules := []*httpserver.Rule{}
+	wildcardHostRules := []*httpserver.Rule{}
+	noHostRules := []*httpserver.Rule{}
```

Comment:
there could be more than one ingress, we need to consider this case also.

Reply:
updated!